FirebaseAuth/Tests/Unit/FIRRevokeTokenResponseTests.m

```diff
@@ -0,0 +1,117 @@
+#import "FirebaseAuth/Tests/Unit/FIRFakeBackendRPCIssuer.h"
+
+/** @var kFakeToken
+    @brief The fake token to use in the test request.
+ */
+static NSString *const kFakeToken = @"fakeToken";
+
+/** @var kFakeIDToken
+    @brief The fake ID token to use in the test request.
+ */
+static NSString *const kFakeIDToken = @"fakeIDToken";
+
+/** @var kFakeToken
+    @brief The fake token to use in the test request.
+ */
+static NSString *const kFakeTokenKey = @"tokenKey";
+
+/** @var kFakeIDToken
+    @brief The fake ID token to use in the test request.
+ */
+static NSString *const kFakeIDTokenKey = @"idTokenKey";
+
+/** @var kFakeAPIKey
+    @brief The fake API key to use in the test request.
+ */
+static NSString *const kFakeAPIKey = @"APIKey";
+
+/** @var kFakeFirebaseAppID
+    @brief The fake Firebase app ID to use in the test request.
+ */
+static NSString *const kFakeFirebaseAppID = @"appID";
+
+/** @var kExpectedAPIURL
+    @brief The expected URL for the test calls.
+ */
+static NSString *const kExpectedAPIURL =
+    @"https://www.googleapis.com/identitytoolkit/v3/relyingparty/revokeToken?key=APIKey";
+
+@interface FIRRevokeTokenResponseTests : XCTestCase
+@end
+
+@implementation FIRRevokeTokenResponseTests {
+  /** @var _RPCIssuer
+      @brief This backend RPC issuer is used to fake network responses for each test in the suite.
+          In the @c setUp method we initialize this and set @c FIRAuthBackend's RPC issuer to it.
+   */
+  FIRFakeBackendRPCIssuer *_RPCIssuer;
+
+  /** @var _requestConfiguration
+      @brief This is the request configuration used for testing.
+   */
+  FIRAuthRequestConfiguration *_requestConfiguration;
+}
+
+- (void)setUp {
+  [super setUp];
+  FIRFakeBackendRPCIssuer *RPCIssuer = [[FIRFakeBackendRPCIssuer alloc] init];
+  [FIRAuthBackend setDefaultBackendImplementationWithRPCIssuer:RPCIssuer];
+  _RPCIssuer = RPCIssuer;
+  _requestConfiguration = [[FIRAuthRequestConfiguration alloc] initWithAPIKey:kFakeAPIKey
+                                                                        appID:kFakeFirebaseAppID];
+}
+
+/** @fn testSuccessfulRevokeTokenResponse
+    @brief Tests a succesful attempt of the verify password flow.
+ */
+- (void)testSuccessfulResponse {
+  FIRRevokeTokenRequest *request =
+      [[FIRRevokeTokenRequest alloc] initWithToken:kFakeToken
+                                           idToken:kFakeIDToken
+                              requestConfiguration:_requestConfiguration];
+  __block BOOL callbackInvoked;
+  __block FIRRevokeTokenResponse *RPCResponse;
+  __block NSError *RPCError;
+  [FIRAuthBackend
+      revokeToken:request
+         callback:^(FIRRevokeTokenResponse *_Nullable response, NSError *_Nullable error) {
+           RPCResponse = response;
+           RPCError = error;
+           callbackInvoked = YES;
+         }];
+
+  [_RPCIssuer respondWithJSON:@{}];
+
+  XCTAssert(callbackInvoked);
+  XCTAssertNotNil(RPCResponse);
+}
+
```

Comment:
Add a test for a nil callback

Reply:
The callback is nonnull. You mean a nil response?

Reply:
The second parameter to revokeToken is nullable. See line 1543 of FIRAuth.m